### PR TITLE
LibJS: Implement 'new.target'

### PR DESCRIPTION
**Pull request by linusg**
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new.target

Also did some cleanup around syntax error line/columns.

### CODE REVIEW

File: Libraries/LibJS/Tests/functions/function-new-target.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            this.newTarget = new.target;
+        }
+    }
+    expect(new baz().newTarget).toEqual(baz);
+});
+
+// FIXME: This does not work as expected as toEval() places the code inside a function :|
+// test("syntax error outside of function", () => {
+//     expect("new.target").not.toEval();
+// });
```
Comment (linusg) [inline Libraries/LibJS/Tests/functions/function-new-target.js:26]: does this violate the no commented-out code rule?! :D

Comment (awesomekling) [reply Libraries/LibJS/Tests/functions/function-new-target.js:26]: Shouldn't this be a skipped test rather than a comment tho?

Comment (linusg) [reply Libraries/LibJS/Tests/functions/function-new-target.js:26]: hah good point. fixed & resolves the question :)

**Comment by awesomekling** [comment]
Nicely implemented!

**Pull request merged**